src/actions/publish.js

```diff
@@ -122,7 +211,16 @@ export function attemptToPublish(): ThunkAction<Promise<boolean>> {
 
       const hash = extractProfileTokenFromJwt(hashOrToken);
 
-      // The previous line was async, check to make sure that this request is still valid.
+      // Because we want to store the published profile even when the upload
+      // generation changed, we store the data here, before the state is fully
+      // updated, and we'll have to predict the state inside this function.
+      await storeJustPublishedProfileData(
```

Comment:
Ah yeah, I was definitely counting on the memoization to save us here, but indeed I'll double check this is the case and try to make it more obvious that we're not running expensive things twice.

Reply:
So, with the patch for #2646, this is much less scary because we only act on the prepublishedState in the whole function.
_But_ there could be cases (maybe now, maybe in the future) where the user could change the "remove profile information" checkboxes _while_ we're uploading. So I'll try to do something like you propose.

Reply:
So I did it only for the sanitization information. I believe other selectors are pretty cheap to run again.